build-tools-internal/src/main/groovy/elasticsearch.run.gradle

```diff
@@ -28,6 +28,18 @@ testClusters.register("runTask") {
       // Not enabled by default in the build, otherwise all integration tests will download the databases from our geoip download service:
       systemProperty 'ingest.geoip.downloader.enabled.default', 'true'
       setting 'xpack.security.enabled', 'true'
+      setting 'xpack.security.transport.ssl.enabled',  'true'
+      setting 'xpack.security.transport.ssl.client_authentication', 'required'
+      extraConfigFile 'transport-tls-keystore.p12', new File(projectDir, "build-tools-internal/src/main/resources/run.ssl/private-cert1.p12")
+      extraConfigFile 'public-ca.pem', new File(projectDir, "build-tools-internal/src/main/resources/run.ssl/public-ca.pem")
+      setting 'xpack.security.transport.ssl.keystore.path', 'transport-tls-keystore.p12'
+      setting 'xpack.security.transport.ssl.certificate_authorities', 'public-ca.pem'
+      if (providers.systemProperty('https').isPresent()) {
```

Comment:
Why use a system property vs an `@Option` on the run task?

Reply:
yeah, that is probably better , will update.

Reply:
@mark-vieira - I remember now why I used a system property ... I want `run` and `run-ccs` to be configured similarly. For `run` I could use `@Option` , but for `run-ccs` I don't think I can gracefully. Since the configuration (and the if condition) belongs to the test cluster configuration and not the `run-ccs` RunTask's configuration (and don't think you can pass these arguments around between tasks).  I could probably add it via the doFirst inside the `run-ccs`RunTask but it's kinda ugly.

Reply:
I'm not sure why the `RunTask` couldn't add that configuration to it's configured test clusters in the `beforeStart` method. This is really not much different than enabling debug.

Reply:
@mark-vieira - I like this approach. I started trying to mirror many of the settings from runTask to the test cluster configuration, but the configuration got weird with odd overlapped/competed with test cluster configuration. So instead, I introduced a simple option `-https` to the run task which will setup the multiple settings for basic https support. Additionally, for transport tls it will auto configure it if no other settings with that prefix are configured, this is similar to default auto configure, but imo simpler to know when it is enabled vs. not and uses static certificates (which makes manual testing easier).  Any more advanced config can be explicitly set by the developer.